testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -5643,7 +5643,7 @@ private List<DataMappingTestSetup> testDataMappingSmokeTestData()
                 .add(new DataMappingTestSetup("timestamp(3) with time zone", "TIMESTAMP '2020-02-12 15:03:00 +01:00'", "TIMESTAMP '9999-12-31 23:59:59.999 +12:00'"))
                 .add(new DataMappingTestSetup("timestamp(6) with time zone", "TIMESTAMP '1969-12-31 15:03:00.123456 +01:00'", "TIMESTAMP '1969-12-31 17:03:00.123456 +01:00'"))
                 .add(new DataMappingTestSetup("timestamp(6) with time zone", "TIMESTAMP '2020-02-12 15:03:00 +01:00'", "TIMESTAMP '9999-12-31 23:59:59.999999 +12:00'"))
-                .add(new DataMappingTestSetup("char(3)", "'ab'", "'zzz'"))
+                .add(new DataMappingTestSetup("char(3)", "'ab '", "'zzz'"))
```

Comment:
`testDataMappingSmokeTest.DataMappingTestSetup("char(3)", "'ab'", "'zzz'")` is failing due to the fact that char is padded upon coercion:

```sql
CREATE TABLE test_data_mapping_smoke_char_3_63hssk8sp8 
    AS SELECT CAST(row_id AS varchar(50)) row_id, CAST(value AS char(3)) value, CAST(value AS char(3)) another_column 
    FROM (VALUES   ('null value', NULL),   ('sample value', 'ab'),   ('high value', 'zzz'))  t(row_id, value);

DESCRIBE test_data_mapping_smoke_char_3_63hssk8sp8;
-- MaterializedResult{rows=[[row_id, varchar, , ], [value, varchar, , ], [another_column, varchar, , ]]}

SELECT value FROM test_data_mapping_smoke_char_3_63hssk8sp8;
-- value='ab '

-- from assertion:
SELECT row_id FROM test_data_mapping_smoke_char_3_63hssk8sp8 WHERE rand() = 42 OR value = 'ab'
-- and expect VALUES 'sample value' -> failure as 'ab' != 'ab '
```

Changing it here breaks it for other connectors:

```
Error:    TestClickHouseConnectorTest>BaseConnectorTest.testDataMappingSmokeTest:5556->BaseConnectorTest.testDataMapping:5590->AbstractTestQueryFramework.assertQuery:350 For query 20240415_123625_02027_285wm: 
 SELECT row_id FROM test_data_mapping_smoke_char_3_mhz2ny6ak4 WHERE rand() = 42 OR value = 'ab '
not equal
Actual rows (up to 100 of 0 extra rows shown, 0 rows in total):
    
Expected rows (up to 100 of 1 missing rows shown, 1 rows in total):
    [sample value]
```

I will use `filterDataMappingSmokeTestData` and change 'ab' -> 'ab ' just for Iceberg tests.